### PR TITLE
add sso hooks

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Support for sso

### CODE REVIEW

**Review by FabienLelaquais — DISMISSED** [review]
I think I see this coming...

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -2609,6 +2609,8 @@ def __register_blueprint(self):
         # server URL Rule for flask rendered react-router
         pages_bp.add_url_rule(f"/{Gui.__INIT_URL}", view_func=self.__init_route)
 
+        _Hooks()._add_external_blueprint(self, __name__)
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py:2630]: 😍

File: taipy/gui/_default_config.py [hunk 2]
```diff
@@ -46,7 +46,7 @@
     "change_delay": None,
     "chart_dark_template": None,
     "base_url": "/",
-    "client_url": "http://localhost:5000",
+    "client_url": "http://127.0.0.1:5000",
```
Comment (FredLL-Avaiga) [inline taipy/gui/_default_config.py]: why not keep localhost everywhere ?

**Review by FredLL-Avaiga — APPROVED** [review]

**Review by FabienLelaquais — APPROVED** [review]
👌

File: taipy/gui/_default_config.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/_default_config.py]: I will want to keep the host as an Ip address for now. localhost will be used by the client_url due to the sso stuff only allowing localhost and not ip addresses in most sso protocols.

**Pull request merged**